### ISSUE TITLE
Rounding

### ISSUE THREAD

**Issue by ariebovenberg** [ariebovenberg/whenever#152]
It'd make sense to be able to round datetimes in different ways:

- by method: (banker's) rounding, ceil, floor
- by unit: seconds, minutes, hours, etc.
- by amount: 15 minutes, 30 seconds, etc.

The temporal API probably gives a big hint for doing this.

**Comment by BurntSushi** [comment]
Yeah, the Temporal API does indeed provide rounding APIs for datetime types. But what's even more useful, IMO, is rounding and re-balancing for durations. There are a lot of options availabe on, for example, [`ZonedDateTime.until`](https://tc39.es/proposal-temporal/docs/zoneddatetime.html#until) that permit rounding the span returned or even choosing the "biggest" unit.

**Comment by ariebovenberg** [comment]
When it comes to re-balancing of durations, this issue is relevant as well: #108.

**Comment by ariebovenberg** [comment]
Rounding has been added in the latest release. For discussion on improving the delta API, see #108
